### ISSUE TITLE
"Source file found twice under different module names" doesn't work in some cases

### ISSUE THREAD

**Issue by bonprosoft** [python/mypy#9058]
I found that "Source file found twice under different module names" error doesn't happen in some cases.

### Files
Same as `case testDuplicateModules` in `test-data/unit/cmdline.test` ([source](https://github.com/python/mypy/blob/8219564d365ba29dda8c4383594d9db91f26feec/test-data/unit/cmdline.test#L1071-L1084)).
```
.
├── mypy.ini
└── src
    ├── a.py
    ├── foo
    │   ├── bar.py
    │   └── __init__.py
    └── __init__.py
```

### Actual behavior
```sh
$ mypy src
src/foo/bar.py: error: Source file found twice under different module names: 'src.foo.bar' and 'foo.bar'
Found 1 error in 1 file (checked 4 source files)

$ mypy `pwd`/src
src/foo/bar.py:1: error: Unsupported operand types for + ("int" and "str")
src/foo/bar.py:1: error: Unsupported operand types for + ("int" and "str")
Found 2 errors in 1 file (checked 4 source files)
```

### Expected behavior

```sh
$ mypy src
src/foo/bar.py: error: Source file found twice under different module names: 'src.foo.bar' and 'foo.bar'
Found 1 error in 1 file (checked 4 source files)

$ mypy `pwd`/src
src/foo/bar.py: error: Source file found twice under different module names: 'src.foo.bar' and 'foo.bar'
Found 1 error in 1 file (checked 4 source files)
```

### Files
Same as `case testDuplicateModules` in `test-data/unit/cmdline.test` except `mypy.ini`.

```
.
├── mypy.ini
└── src
    ├── a.py
    ├── foo
    │   ├── bar.py
    │   └── __init__.py
    └── __init__.py
```

- mypy.ini
```
[mypy]
mypy_path = /absolute/path/to/src
```

### Actual behavior
```sh
$ mypy src
src/foo/bar.py:1: error: Unsupported operand types for + ("int" and "str")
src/foo/bar.py:1: error: Unsupported operand types for + ("int" and "str")
Found 2 errors in 1 file (checked 4 source files)

$ mypy `pwd`/src
src/foo/bar.py: error: Source file found twice under different module names: 'src.foo.bar' and 'foo.bar'
Found 1 error in 1 file (checked 4 source files)
```

### Expected behavior

```sh
$ mypy src
src/foo/bar.py: error: Source file found twice under different module names: 'src.foo.bar' and 'foo.bar'
Found 1 error in 1 file (checked 4 source files)

$ mypy `pwd`/src
src/foo/bar.py: error: Source file found twice under different module names: 'src.foo.bar' and 'foo.bar'
Found 1 error in 1 file (checked 4 source files)
```

## Versions
```sh
$ mypy --version
mypy 0.790+dev.8219564d365ba29dda8c4383594d9db91f26feec
$ python --version
Python 3.6.10
```

**Comment by JukkaL** [comment]
Once #9092 has been merged, could you please post the output when you run `mypy` with `-v`? (Feel to trim out things from the output that don't seem relevant.)

**Comment by JukkaL** [comment]
Ah I didn't notice that there's already a fix. No need do further debugging if you've already figured this out!

**Comment by bonprosoft** [comment]
I am very sorry for confusing you.
As you mentioned, I have already submitted a PR to fix the problem in #9059.
